### PR TITLE
US-33: Replay

### LINKED ISSUE #62: US-33 Create a replay view fragment
- [x] Create fragment
- [x] Add a map view
- [x] React to control changes
- [x] Handle location events
- [x] Handle catch events
- [ ] Use an online tile downloader
- [ ] Test the shit

### LINKED ISSUE #63: US-33 Create a replay control fragment
Currently blocked by firebase setup and dagger investigations

- [x] Create the fragment
- [x] Design the view
- [x] Notify the replay view about changes
- [x] Handle time change
- [x] Play the timeline
- [x] Handle speed change
- [x] Test it

### LINKED ISSUE #64: US-33 Create a replay activity
Currently blocked by firebase setup and dagger investigations

- [x] Create the activity
- [x] Load the replay file
- [x] Parse the replay file
- [x] Prepare the game history based on the replay file
- [x] Initialize the fragments
- [x] Test it

### LINKED ISSUE #80: US-33 Replay selection activity
- [x] Create the activity
- [x] Display the list of know replays from SQLite
- [x] Refresh on swipe (query firebase)
- [x] If a replay is not on the local storage, signal it visually
- [x] If a replay is not on the local storage, propose to download it from firebase storage upon click
- [x] Start the replay activity on replay selection with the replay id (or path?)
- [x] Test it

### PR DESCRIPTION
**Pull request by Zorvalt**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/replay/IReplayRepository.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package ch.epfl.sdp.replay
+
+import ch.epfl.sdp.db.Callback
+import ch.epfl.sdp.game.data.Game
+import java.io.File
+
+interface IReplayRepository {
+    fun getAllGames(cb: Callback<List<ReplayInfo>>)
```
Comment (pauldms) [inline app/src/main/java/ch/epfl/sdp/replay/IReplayRepository.kt:8]: I think you should fetch all games for a specific user, not all the games in the database, so getAllGames might need a uid parameter

File: app/src/main/java/ch/epfl/sdp/replay/ReplayInfo.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package ch.epfl.sdp.replay
```
Comment (pauldms) [inline app/src/main/java/ch/epfl/sdp/replay/ReplayInfo.kt:1]: this class might belong in the data package

File: app/src/main/java/ch/epfl/sdp/replay/ReplayInfoListFragment.kt [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        (activity?.applicationContext as HideAndHuntApplication).appComponent.inject(this)
+        super.onCreate(savedInstanceState)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        _binding = FragmentReplayInfoListBinding.inflate(inflater)
+
+        repo.getAllGames { games ->
```
Comment (pauldms) [inline app/src/main/java/ch/epfl/sdp/replay/ReplayInfoListFragment.kt:36]: again, this does not fetch the games only for the current user

**Review by pauldms — APPROVED** [review]
Great work, might need to discuss some things with the repo interfaces. This PR is very long, try to divide and merge smaller PRs next time.

**Review by ProtectedVariable — APPROVED** [review]
same as paul

**Pull request merged**